src/test/scala/is/hail/methods/LDPruneSuite.scala

```diff
@@ -323,4 +327,66 @@ class LDPruneSuite extends SparkSuite {
     val prunedVDS = LDPrune(filteredVDS, nCores, r2Threshold = 1, windowSize = 0, memoryPerCoreMB = 200)
     assert(prunedVDS.countRows() == filteredVDS.countRows())
   }
+
+  @Test def testGetIntervals() {
+    val rows = List[(Long, String, Int)](
+      (0, "X", 5),
+      (1, "X", 7),
+      (2, "X", 13),
+      (3, "X", 14),
+      (4, "X", 65),
+      (5, "X", 70),
+      (6, "X", 73),
+      (7, "Y", 74),
+      (8, "Y", 75),
+      (9, "Y", 200),
+      (10, "Y", 300)
+    ).map { case (x, y, z) => Row(x, y, z) }.toIndexedSeq
+    val tbl = Table.parallelize(hc, rows, TStruct("index" -> TInt64(), "chromosome" -> TString(), "position" -> TInt32()),
+      IndexedSeq[String](), None)
+    val intervals = LDPruneSparseEntriesTable.windows(tbl, 10)
+    val expected = List((7, 8), (4, 6), (0, 3))
```

Comment:
I'm confused about what `windows` is expected to return. What happened to variants 9 and 10? Just because they each lie in their own window doesn't mean we want to discard them, right? They are supremely independent. :) I'd also like to see a more complete test that shows overlapping windows in the output.

Reply:
I'm collecting correlated variants here so that I can discard them (using a maximal independent set) during LD Prune. 9 and 10 would still be in the dataset, because I filter to remove variants, not to keep them. Do you need a method that keeps 9 and 10? Otherwise, it's simpler for me to exclude them.

With regard to overlapping windows, I'm coalescing the intervals in the method so that there are no overlapping windows.

Reply:
I see, so I didn't understand what your algorithm was doing. I'll take another look once you've addressed the other comments.